src/Domain/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Codidact.Domain.Extensions
+{
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Converts a string to snake_case
+        /// </summary>
+        /// <param name="input">The string to convert</param>
+        /// <returns></returns>
+        public static string ToSnakeCase(this string input)
+        {
+            if (string.IsNullOrEmpty(input)) { return input; }
+
+            var startUnderscores = Regex.Match(input, @"^_+");
+            return startUnderscores + Regex.Replace(input, @"([a-z0-9])([A-Z])", "$1_$2").ToLower();
```

Comment:
non-merge-blocking: you don't need the parens around your character classes in this regex. `@"[a-z0-9][A-Z]"` works, or if you want to keep the parens for clarity, you can make them non-capturing groups: `@"(?:[a-z0-9])(?:[A-Z])"`

Reply:
The unit test returns `came$1_$2as$1_$2xample` for the two options you posted

Reply:
Gah, missed that you were using the captures. Yup, this is all good.